### PR TITLE
Optional formatters

### PR DESCRIPTION
**Pull request by adnanhemani**
[*] Wrote test for feature
[*] Added changes in the Changelog section in README.md
[*] Bumped version number (delete if unneeded)

**Changes proposed:**
added optional formatters for with_column, with_columns, append_column

### CODE REVIEW

File: datascience/tables.py [hunk 1]
```diff
@@ -1585,30 +1590,30 @@ def with_column(self, label, values):
         ValueError: Column length mismatch. New column does not have the same number of rows as table.
         """
         new_table = self.copy()
-        new_table.append_column(label, values)
+        new_table.append_column(label, values, formatter)
         return new_table
 
-    def with_columns(self, *labels_and_values):
+    def with_columns(self, *labels_and_values_and_formats):
```
Comment (papajohn) [inline datascience/tables.py]: I think the right signature is 

    def with_columns(self, *labels_and_values, *, formatter=None)

and it's limited to giving the same formatter to every new column or no formatter at all.

File: datascience/tables.py [hunk 2]
```diff
@@ -426,6 +426,7 @@ def append_column(self, label, values):
 
                 If a list or array, the new column contains the values in
                 ``values``, which must be the same length as the table.
+            ``formatter`` (single formatter): Adds a formatter to the column being appended. No formatter added by default.
```
Comment (papajohn) [inline datascience/tables.py]: Please wrap comments at 80 characters to be consistent.

File: README.md [hunk 3]
```diff
@@ -27,6 +27,9 @@ pip install datascience
 
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+### v0.10.0
```
Comment (papajohn) [inline README.md]: Probably 0.9.1, even though it's not quite semantic versioning.

**Comment by adnanhemani** [comment]
Requested changes have been made. Please re-review!

File: README.md [hunk 4]
```diff
@@ -27,6 +27,10 @@ pip install datascience
 
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+
+### v0.9.6
+- Added optional formatters to "Table.with_column", "Table.with_columns", and "Table.append_column".  
+
```
Comment (davidwagner) [inline README.md]: Let's remember to update the version before merging.

File: datascience/tables.py [hunk 5]
```diff
@@ -1610,34 +1613,38 @@ def with_column(self, label, values, *rest):
...
+    def with_columns(self, *labels_and_values, **formatter):
         """Return a table with additional or replaced columns.
 
 
         Args:
-            ``labels_and_values``: An alternating list of labels and values or
-                a list of label-value pairs. If one of the labels is in
+            ``labels_and_values_and_formats``: An alternating list of labels, values, and (optionally) formatters or
+                a list of label-value pairs and/or label-value-formatter trios. If one of the labels is in
                 existing table, then every value in the corresponding column is
```
Comment (davidwagner) [inline datascience/tables.py]: I find this confusing.  Is there a way to explain this more clearly?  Can we add an example?

Comment (davidwagner) [inline datascience/tables.py]: There is no argument named `labels_and_values_and_formats`.

Comment (adnanhemani) [reply datascience/tables.py]: Oops, sorry. Made the correction!

Comment (adnanhemani) [reply datascience/tables.py]: Adding one now!

File: datascience/tables.py [hunk 6]
```diff
@@ -1599,29 +1602,32 @@ def with_column(self, label, values, *rest):
...
                 every row of corresponding column takes on that value.
+            ''formatter'' (single Formatter value): A single formatter value that will be applied to all columns 
+                being added using this function call.
 
         Raises:
             ``ValueError``: If
                 - any label in ``labels_and_values`` is not a valid column
                     name, i.e if label is not of type (str).
-                - if any value in ``labels_and_values`` is a list/array and
+                - if any value in ``labels_and_values`` is a list/array and`
```
Comment (davidwagner) [inline datascience/tables.py]: Is that extra backtick extraneous?

File: datascience/tables.py [hunk 7]
```diff
@@ -1599,29 +1602,32 @@ def with_column(self, label, values, *rest):
...
         Args:
-            ``labels_and_values``: An alternating list of labels and values or
-                a list of label-value pairs. If one of the labels is in
-                existing table, then every value in the corresponding column is
+            ``labels_and_values``: An alternating list of labels and values or a list of label-value pairs. 
+                If one of the labels is in existing table, then every value in the corresponding column is
                 set to that value. If label has only a single value (``int``),
                 every row of corresponding column takes on that value.
+            ''formatter'' (single Formatter value): A single formatter value that will be applied to all columns 
+                being added using this function call.
```
Comment (davidwagner) [inline datascience/tables.py]: These should be line-wrapped to 80 columns.

File: datascience/tables.py (hunk 6 above)
Comment (adnanhemani) [reply datascience/tables.py]: Wow, what a catch! Yes, it was - pushing the changes now!

**Pull request merged**